### PR TITLE
Protect JITServer code with `#if defined(J9VM_OPT_JITSERVER)`

### PR DESCRIPTION
**Pull request by mpirvu**
This fixes a build break on platforms that don't have JITServer enabled by default.

### CODE REVIEW

**Comment by mpirvu** [comment]
jenkins compile all jdk17

**Comment by keithc-ca** [comment]
Building with `--disable-jitserver` I found two other files that won't compile:
```
runtime/compiler/control/CompilationThread.cpp
runtime/vm/jvminit.c
```

**Comment by keithc-ca** [comment]
I aborted the builds.

**Comment by mpirvu** [comment]
@keithc-ca The code that I just pushed should solve the compile issue.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins compile amac jdk17

**Comment by keithc-ca** [comment]
I don't think we need to wait for approval from @dsouzai.

**Pull request merged**
